### PR TITLE
DO NOT MERGE: Test adding symbolset support to SwiftGen

### PR DESCRIPTION
**Pull request by hisaac**
### Short description 📝

**DO NOT MERGE THIS. I am opening this pull request to the Tuist repo just to run CI builds and tests, to verify nothing breaks with the changes to that dependency.**

I've opened a pull request to @fortmarek's fork of SwiftGen that is used in this project, which pulls in changes from SwiftGen's develop branch that add support for symbolset assets (https://github.com/fortmarek/SwiftGen/pull/1).

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/). (in progress)
- [ ] ~~The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.~~ (changes not intended to be merged)
- [ ] ~~In case the PR introduces changes that affect users, the documentation has been updated.~~ (changes not intended to be merged)

### CODE REVIEW

**Comment by fortmarek** [comment]
Hey, @hisaac!

Thanks for doing this. Could you maybe try to point instead of a fork to the main repo? There were some issues we were having with plist generation but hopefully, those have been resolved by now.

File: Package.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ let package = Package(
         .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMajor(from: "0.4.3")),
         .package(url: "https://github.com/marmelroy/Zip.git", .upToNextMinor(from: "2.1.1")),
         .package(url: "https://github.com/tuist/GraphViz.git", .branch("tuist")),
-        .package(url: "https://github.com/fortmarek/SwiftGen", .branch("stable")),
+        .package(url: "https://github.com/hisaac/SwiftGen", .branch("add-symbolset-support")),
```
Comment (pepibumur) [inline Package.swift:67]: What prevents us from pushing those changes upstream? Depending on a branch of a fork doesn't seem a good idea.

Comment (hisaac) [reply Package.swift:67]: I opened this pull request just to run CI with the changes that are on my fork. The intent is to to get the changes merged into @fortmarek's fork eventually, but I wanted to test the changes within Tuist before we do that.

**Comment by hisaac** [comment]
> Hey, @hisaac!
> 
> Thanks for doing this. Could you maybe try to point instead of a fork to the main repo? There were some issues we were having with plist generation but hopefully, those have been resolved by now.

I opened this draft pull request just to run CI on the changes that are in my fork of the SwiftGen repo. I'm doing this just to run tests — this change is not meant to be merged in.

If [my pull request to your SwiftGen fork](https://github.com/fortmarek/SwiftGen/pull/1) get merged in, then there would be no change necessary to Tuist itself, it will start getting the updated SwiftGen changes automatically because it's pointing to a branch here in the `Project.swift` and `Package.swift` files.

**Comment by fortmarek** [comment]
> I opened this draft pull request just to run CI on the changes that are in my fork of the SwiftGen repo. I'm doing this just to run tests — this change is not meant to be merged in.

I see - I'd still try to point to the main repo rather than my fork which was supposed to be only temporary solution.

**Comment by hisaac** [comment]
> I see - I'd still try to point to the main repo rather than my fork which was supposed to be only temporary solution.

@fortmarek ohhhh, I believe this is a misunderstanding. When you were saying "main", I assumed you meant `fortmarek/SwiftGen`, and that "fork" meant `hisaac/SwiftGen`.

I think I understand now, that you're hoping the additions you made in `fortmarek/SwiftGen` are no longer needed, and that we can just point to `SwiftGen/SwiftGen` now instead. Is that right?

I'll make that change, and we'll see how it goes 👍

And thanks for helping me along as I learn how Tuist and your development process works. 🙏

**Comment by fortmarek** [comment]
> I think I understand now, that you're hoping the additions you made in fortmarek/SwiftGen are no longer needed, and that we can just point to SwiftGen/SwiftGen now instead. Is that right?

Yes, that's correct, sorry for not being clear enough! 🤞

**Comment by fortmarek** [comment]
Let me know if you run into any issues, I might have some context that can help you here.

**Comment by hisaac** [comment]
So I updated the `Package.swift` and `Project.swift` files to point at the revision that includes the symbolset support, but I get some dependency resolution errors. 😞

Here's the output in my terminal when I run `./fourier generate tuist`:

```shell
❯ ./fourier generate tuist
Updating https://github.com/fortmarek/swifter.git
Updating https://github.com/tuist/GraphViz.git
Updating https://github.com/SwiftGen/SwiftGen
Updating https://github.com/apple/swift-log.git
Updating https://github.com/ReactiveX/RxSwift.git
Updating https://github.com/kylef/PathKit.git
Updating https://github.com/FabrizioBrancati/Queuer.git
Updating https://github.com/tuist/BlueSignals.git
Updating https://github.com/krzyzanowskim/CryptoSwift.git
Updating https://github.com/kylef/Spectre.git
Updating https://github.com/tadija/AEXML
Updating https://github.com/SwiftGen/StencilSwiftKit.git
Updating https://github.com/apple/swift-argument-parser.git
Updating https://github.com/CombineCommunity/CombineExt.git
Updating https://github.com/pointfreeco/combine-schedulers
Updating https://github.com/stencilproject/Stencil.git
Updating https://github.com/apple/swift-tools-support-core.git
Updating https://github.com/shibapm/PackageConfig.git
Updating https://github.com/tuist/XcodeProj.git
Updating https://github.com/marmelroy/Zip.git
Updating https://github.com/rnine/Checksum.git
Updating https://github.com/kishikawakatsumi/KeychainAccess.git
Updating https://github.com/shibapm/Komondor.git
Updating https://github.com/JohnSundell/ShellOut.git
Updating https://github.com/tid-kijyun/Kanna.git
Updating https://github.com/pointfreeco/xctest-dynamic-overlay
Updating https://github.com/jpsim/Yams.git
Updating https://github.com/kylef/Commander.git
error: Dependencies could not be resolved because root depends on 'Stencil' 0.14.0..<1.0.0.
'Stencil' >= 0.14.0 cannot be used because 'Stencil' >= 0.14.0 depends on 'PathKit' 1.0.0..<2.0.0 and root depends on 'PathKit' 0.9.0..<1.0.0.
error: ExitCode(rawValue: 1)
[0/1] Planning buildwarning: failed to load the cached build description: fatalError
Generating workspace Tuist.xcworkspace
Generating project Tuist
Resolving package dependencies using xcodebuild
xcodebuild: error: Could not resolve package dependencies:
  Dependencies could not be resolved because root depends on 'PathKit' 1.0.0..<2.0.0 and root depends on 'PathKit' 0.9.0..<1.0.0.


The 'xcodebuild' command exited with error code 74 and message:
xcodebuild: error: Could not resolve package dependencies:
  Dependencies could not be resolved because root depends on 'PathKit' 1.0.0..<2.0.0 and root depends on 'PathKit' 0.9.0..<1.0.0.


Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

I tried changing the `PathKit` dependency from `.upToNextMajor(from: "1.0.0")),` to ``.upToNextMajor(from: "2.0.0")),`, but got other dependency errors.

Any ideas on how to resolve things? Perhaps the changes in your fork are still needed after all?

**Comment by fortmarek** [comment]
Yeah, the issue is with `PathKit`. Maybe [this](https://github.com/kylef/PathKit/pull/80) will get them to bump it? Because this has already been raised there but [dismissed](https://github.com/SwiftGen/SwiftGen/issues/819). Might be worth creating a PR but let's keep pointing to the fork for now.

**Comment by hisaac** [comment]
Sounds good @fortmarek. I've pushed the changes here for testing.

**Comment by hisaac** [comment]
@fortmarek The `./fourier lint lockfiles` step failed because the Kanna dependency is showing 2 different values between `Package.resolved` and `.packages.resolved`. Is this a true issue? Is it something that would need to be addressed in this test pull request, or are the rest of the test runs passing enough to consider this a success?

**Comment by fortmarek** [comment]
> Is this a true issue? Is it something that would need to be addressed in this test pull request, or are the rest of the test runs passing enough to consider this a success?

I think for the purposes of this PR it should be fine. For a normal PR, they do need to be the same.

**Comment by hisaac** [comment]
The changes in https://github.com/fortmarek/SwiftGen/pull/1 have now been merged, so this pull request is no longer needed. Thanks for the help everyone.

**Pull request closed without merging**